lib/prebundler/s3_backend.rb

```diff
@@ -1,16 +1,18 @@
+# frozen_string_literal: true
+
 require 'aws-sdk'
 
 module Prebundler
   class S3Backend
-    attr_reader :access_key_id, :secret_access_key, :bucket, :region, :endpoint, :force_path_style
+    attr_reader :access_key_id, :secret_access_key, :bucket, :region
 
     def initialize(options = {})
-      @access_key_id = options.fetch(:access_key_id)
-      @secret_access_key = options.fetch(:secret_access_key)
-      @bucket = options.fetch(:bucket)
-      @region = options.fetch(:region) { 'us-east-1' }
-      @endpoint = options.fetch(:endpoint) { 's3.amazonaws.com' }
-      @force_path_style = options.fetch(:force_path_style) { false }
+      @bucket             = options.fetch(:bucket)
+
+      @client             = options.fetch(:client, nil)
```

Comment:
If we're passing the client in, do we need to continue passing the other options too (i.e. `access_key_id` and friends)?

Reply:
Never mind, this allows not only backwards-compatibility but also makes standard setups easy.